okhttp-android/src/main/kotlin/okhttp3/android/NetworkSelection.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.android
+
+import android.net.Network
+import android.os.Build
+import androidx.annotation.RequiresApi
+import javax.net.SocketFactory
+import okhttp3.ExperimentalOkHttpApi
+import okhttp3.OkHttpClient
+import okhttp3.android.internal.NetworkPinInterceptor
+
+@ExperimentalOkHttpApi
+object NetworkSelection {
+  @RequiresApi(Build.VERSION_CODES.Q)
+  fun OkHttpClient.Builder.withNetwork(network: Network?): OkHttpClient.Builder {
```

Comment:
I like this part of the API.

What happens if we attempt this in the main OkHttp module? We’d have a runtime dependency on `Network`, but you wouldn’t call this API if you don’t have it

Reply:
I am ok with either. 

I guess ultimately we should choose either

a) we build android specific bits in main module, but you just don't use them if not needed.
b) we encourage every android app to include okhttp and okhttp-android and use those bits.

I'm not sure.